### PR TITLE
gossa: init at 0.2.2

### PR DESCRIPTION
**Pull request by dsymbol**
###### Description of changes
Added the [gossa](https://github.com/pldubouilh/gossa) fast and simple multimedia fileserver.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -4010,6 +4010,11 @@
     githubId = 1931963;
     name = "David Sferruzza";
   };
+  dsymbol = {
+    name = "dsymbol";
+    github = "dsymbol";
+    githubId = 88138099;
+  };
```
Comment (azahi) [inline maintainers/maintainer-list.nix:4017]: Adding yourself as a maintainer should be a separate commit.

File: pkgs/applications/networking/gossa/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  version = "0.2.2";
+
+  goPackagePath = "github.com/pldubouilh/gossa";
+
+  src = fetchgit {
+    url = "https://github.com/pldubouilh/gossa";
+    rev = "v${version}";
+    sha256 = "sha256-eMO9aoI+otGQcvBUJtxciQ7yhUidYizLrDjMVchH3qA=";
+    fetchSubmodules = true;
+  };
```
Comment (azahi) [inline pkgs/applications/networking/gossa/default.nix:18]: Please use `fetchFromGitHub`.

File: pkgs/applications/networking/gossa/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  meta = with lib; {
+    homepage = "https://github.com/pldubouilh/gossa";
+    description = "A fast and simple multimedia fileserver ";
```
Comment (azahi) [inline pkgs/applications/networking/gossa/default.nix:22]:
```suggestion
    description = "A fast and simple multimedia fileserver";
```

File: pkgs/applications/networking/gossa/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
+{ stdenv
+, lib
+, fetchgit
+, buildGoPackage
+}:
+
+buildGoPackage rec {
```
Comment (azahi) [inline pkgs/applications/networking/gossa/default.nix:7]: It's preferred to use `buildGoModule` if possible.

**Review by azahi — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/gossa/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  src = fetchFromGitHub {
+    rev = "v${version}";
+    owner = "pldubouilh";
+    repo = "gossa";
+    sha256 = "sha256-eMO9aoI+otGQcvBUJtxciQ7yhUidYizLrDjMVchH3qA=";
+    fetchSubmodules = true;
+  };
+
+  vendorHash = null;
+  doCheck = false;
```
Comment (azahi) [inline pkgs/applications/networking/gossa/default.nix:20]: Why is this disabled? If there's a reason, you should include it in the comment.

File: pkgs/applications/networking/gossa/default.nix [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  version = "0.2.2";
+
+  src = fetchFromGitHub {
+    rev = "v${version}";
+    owner = "pldubouilh";
+    repo = "gossa";
+    sha256 = "sha256-eMO9aoI+otGQcvBUJtxciQ7yhUidYizLrDjMVchH3qA=";
+    fetchSubmodules = true;
+  };
+
```
Comment (azahi) [inline pkgs/applications/networking/gossa/default.nix:18]: #177698
```suggestion

ldflags = [ "-s" "-w" ];

```

**Review by azahi — CHANGES_REQUESTED** [review]
Looks good, only one thing left.

**Review by azahi — CHANGES_REQUESTED** [review]
Apparently the tests were failing because we can't run them in a sandboxed environment. Can you please replace your derivation with this [file](https://gist.github.com/azahi/72a4fe518a92cf7b718e6338c24e453c). I think I handled all issues in there.

**Comment by dsymbol** [comment]
Seemed to be working for me previously. However, there was a discussion on the Nix discourse regarding network tests not working properly in the sandboxed environment. Thank you for taking the time to review this PR multiple times. :)

**Review by azahi — APPROVED** [review]

**Pull request merged**